packages/cdktf/lib/tokens/private/resolve.ts

```diff
@@ -109,6 +109,13 @@ export function resolve(obj: any, options: IResolveOptions): any {
   // string - potentially replace all stringified Tokens
   //
   if (typeof obj === "string") {
+    // If this is a "list element" Token, it should never occur by itself in string context
+    if (TokenString.forListToken(obj).test()) {
+      throw new Error(
```

Comment:
I almost wonder if we shouldn't just make it to work instead of erroring. If the user tries to use something other than index `0` it will error out anyways, so this could maybe be a nice shortcut.

Reply:
I wonder if we could mask confusing errors here if we don't error 🤔 Wdyt @ansgarm @skorfmann ?

Reply:
hm, how would the native error look like?

Reply:
I'm currently lacking the context to even understand the problem here 😅

Reply:
The idea would be to resolve the token rather than error. The risk I see is that if a user was iterating through an array, they might think it was working, but in reality it wouldn't be since the token array always has length 1 and index 0 would function (if we made it work). Should probably go back to the issue that I believe inspired the error to help make that call.

Reply:
I think I'd go here with the earliest error possible so folks don't think they can traverse an array and can do so without an error and just the unintended HCL.json